### PR TITLE
PROGRAMMES-5953 Change on-now controller to always return a 200

### PR DESCRIPTION
**Pull request by Tasiobg**
Change on-now controller to always return a 200 status code even when no broadcasts are found and set the cache-control to 300 seconds

### CODE REVIEW

File: src/Controller/SchedulesOnNowController.php [hunk 1]
```diff
@@ -33,7 +33,10 @@ public function __invoke(
 
         $broadcast = $broadcastsService->findOnNowByService($network->getDefaultService());
         if (!$broadcast) {
-            throw $this->createNotFoundException('No broadcasts found.');
+            $response = $this->response();
+            $response->setStatusCode(200);
+            $response->setPublic()->setMaxAge(300);
+            return $response;
```
Comment (votemike) [inline src/Controller/SchedulesOnNowController.php]: So, what does this return?

File: src/Controller/SchedulesOnNowController.php [hunk 2]
```diff
@@ -33,7 +33,10 @@ public function __invoke(
 
         $broadcast = $broadcastsService->findOnNowByService($network->getDefaultService());
         if (!$broadcast) {
-            throw $this->createNotFoundException('No broadcasts found.');
+            $response = $this->response();
+            $response->setStatusCode(200);
+            $response->setPublic()->setMaxAge(300);
```
Comment (BPScott) [inline src/Controller/SchedulesOnNowController.php]: The status is 200 by default, and setPublic() is already called in the constructor.All the setters on $response are a fluent interface so I believe this should suffice:

```
return $this->response()->setMaxAge(300);
```

Comment (Tasiobg) [reply src/Controller/SchedulesOnNowController.php]: thanks!

**Comment by programmes-developers** [comment]
Can one of the admins verify this patch?

File: src/Controller/SchedulesOnNowController.php (hunk 1 above)
Comment (Tasiobg) [reply src/Controller/SchedulesOnNowController.php]: this return a 200 response with no content

Comment (votemike) [reply src/Controller/SchedulesOnNowController.php]: Should it be a 204 instead?

Comment (Tasiobg) [reply src/Controller/SchedulesOnNowController.php]: Good point.
204 is usefull for APIs responses, in our case is an AJAX response and I'm not completly sure if using 204 is an improvement or not, seems a bit more clear to understand that the response was empty on purpose.
Anyone else wants to share their views?

Comment (votemike) [reply src/Controller/SchedulesOnNowController.php]: Well, this is loaded by Javascript. How can you tell if it failed or just returned blank content?

Comment (Tasiobg) [reply src/Controller/SchedulesOnNowController.php]: After having a chat with bjorn the 204 looks technically more correct but he is not too sure how the varnish layer will handle the 204 codes so the advice is to leave it as 200

**Pull request merged**